blocks/bricks/__init__.py

```diff
@@ -158,7 +159,56 @@ def _push_initialization_config(self):
                     child.biases_init = self.biases_init
 
 
-class Linear(Initializable):
+class Feedforward(Brick):
+    """Declares an interface for bricks with one input and one output.
+
+    Many bricks have just one input and just one output (activations,
+    :class:`Linear`, :class:`MLP`). To make such bricks interchangable
+    in most contexts they should share an interface for configuring
+    their input and output dimensions. This brick provides such an
+    interface.
+
+    """
+    @abstractproperty
+    def input_dim():
+        """The input dimension of the brick."""
+        pass
+
+    @abstractproperty
+    def output_dim():
+        """The output dimension of the brick."""
+        pass
+
+
+def override_with_attribute(property_, attribute=None):
```

Comment:
I want to give an error if the author of the class forgot to override the properties. E.g. the `MLP` bricks currently redirects the `input_dim` and `output_dim` requests to its `dims` array. If you remove these implementations, you will get an error from accessing an abstract property. Not only when reading, but also at writing, unlike your implementation above.